rmw_fastrtps_shared_cpp/src/rmw_wait_set.cpp

```diff
@@ -80,7 +80,7 @@ __rmw_destroy_wait_set(const char * identifier, rmw_wait_set_t * wait_set)
   RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
     wait set handle,
     wait_set->implementation_identifier, identifier,
-    return RMW_RET_ERROR)
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION)
 
   auto result = RMW_RET_OK;
   auto wait_set_info = static_cast<CustomWaitsetInfo *>(wait_set->data);
```

Comment:
@ahcorde I think this fini function needs a refactor. The two checks right below make no sense for the same reasons exposed above. And I'd go as far as to say that those if clauses below make no sense either -- if you got this far in the function and one of those pointers is `nullptr` something's very wrong.